src/sparseml/core/recipe/recipe.py

```diff
@@ -424,6 +424,58 @@ def dict(self, *args, **kwargs) -> Dict[str, Any]:
 
         return dict_
 
+    def yaml(self, file_path: Optional[str] = None) -> str:
+        """
+        Return a yaml string representation of the recipe.
+
+        :param file_path: optional file path to save yaml to
+        :return: The yaml string representation of the recipe
+        """
+        file_stream = None if file_path is None else open(file_path, "w")
+        yaml_dict = self._get_yaml_dict()
+
+        ret = yaml.dump(
```

Comment:
no conflict between import yaml and your def yaml?

Reply:
ah i see its inside a class

Reply:
Yes thou art correct!